app/controllers/api/v1/conversations_controller.rb

```diff
@@ -0,0 +1,87 @@
+# frozen_string_literal: true
+
+module API
+  module V1
+    # Controller class that is responsible for handling conversation requests.
+    class ConversationsController < ActionController::API
+      include DeviseTokenAuth::Concerns::SetUserByToken
+      include Pundit
+
+      before_action :authenticate_user!
+      before_action :conversation, only: %i[show destroy update add_user delete_user]
+      before_action :authorize_conversation!
+      after_action :verify_authorized
+
+      def index
+        @conversations = current_user.conversations.all
+        render json: @conversations, status: :ok
+      end
+
+      def show
+        if @conversation
+          render json: @conversation, status: :ok
+        else
+          render json: @conversation.errors, status: :not_found
+        end
+      end
+
+      def add_user
+        @conversation.users << User.where(id: params[:user_id]) if params[:user_id].present?
```

Comment:
You should use `.find_by(id: params[:user_id])`
This part also missing checks if users are present.
We should assign user to the conversation only if it exists.
Make sure to update the policy for conversation.